Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenStick.cs

```diff
@@ -13,47 +13,40 @@ public class OnScreenStick : OnScreenControl, IPointerDownHandler, IPointerUpHan
     {
         private void Start()
         {
-            m_StartPos = transform.position;
+            m_StartPos = (transform as RectTransform).anchoredPosition;
+            Canvas canvas = GetComponentInParent<Canvas>();
+            if (canvas)
+                m_Camera = canvas.worldCamera;
         }
 
         public void OnPointerDown(PointerEventData data)
         {
+            RectTransformUtility.ScreenPointToLocalPointInRectangle(GetComponentInParent<RectTransform>(), data.position, m_Camera, out m_PointerDownPos);
         }
 
         public void OnDrag(PointerEventData data)
         {
-            var newPos = Vector2.zero;
+            Vector2 position;
+            RectTransformUtility.ScreenPointToLocalPointInRectangle(GetComponentInParent<RectTransform>(), data.position, m_Camera, out position);
+            Vector2 delta = position - m_PointerDownPos;
 
-            ////REVIEW: is this Y up?
-
-            ////REVIEW: this doesn't make sense; data.position and transform.position are not in the same coordinate space
-
-            var deltaX = (int)(data.position.x - m_StartPos.x);
-            deltaX = Mathf.Clamp(deltaX, -movementRange, movementRange);
-            newPos.x = deltaX;
-
-            var deltaY = (int)(data.position.y - m_StartPos.y);
-            deltaY = Mathf.Clamp(deltaY, -movementRange, movementRange);
-            newPos.y = deltaY;
-
-            ////FIXME: this is setting up a square movement space, not a radial one; relies on normalization on the control to work
-
-            newPos.x /= movementRange;
-            newPos.y /= movementRange;
+            delta = Vector2.ClampMagnitude(delta, movementRange);
+            (transform as RectTransform).anchoredPosition = m_StartPos + (Vector3)delta;
 
+            var newPos = new Vector2(delta.x / movementRange, delta.y / movementRange);
             SendValueToControl(newPos);
-
-            transform.position = new Vector3(m_StartPos.x + newPos.x, m_StartPos.y + newPos.y, m_StartPos.z);
         }
 
         public void OnPointerUp(PointerEventData data)
         {
-            transform.position = m_StartPos;
+            (transform as RectTransform).anchoredPosition = m_StartPos;
```

Comment:
Use direct cast `((RectTransform)transform)` when the cast mustn't fail. Otherwise you end up with an NullReferenceException instead of an InvalidCastException when things go wrong.